doc/usage/extensions/apidoc.rst

```diff
@@ -0,0 +1,126 @@
+.. _ext-apidoc:
+
+:mod:`sphinx.ext.apidoc` -- Generate API documentation from Python packages
+===========================================================================
+
+.. py:module:: sphinx.ext.apidoc
+   :synopsis: Generate API documentation from Python modules
+
+.. index:: pair: automatic; documentation
+.. index:: pair: generation; documentation
+.. index:: pair: generate; documentation
+
+.. versionadded:: 8.2
+
+.. role:: code-py(code)
+   :language: Python
+
+:mod:`sphinx.ext.apidoc` is a tool for automatic generation
+of Sphinx sources from Python packages.
+It provides the :program:`sphinx-apidoc` command-line tool as an extension,
+allowing it to be run during the Sphinx build process.
+
+The extension writes generated source files to a provided directory,
+which are then read by Sphinx using the :mod:`sphinx.ext.autodoc` extension.
+
+.. warning::
+
+   :mod:`sphinx.ext.apidoc` generates source files that
+   use :mod:`sphinx.ext.autodoc` to document all found modules.
+   If any modules have side effects on import,
+   these will be executed by ``autodoc`` when :program:`sphinx-build` is run.
+
+   If you document scripts (as opposed to library modules),
+   make sure their main routine is protected by
+   an ``if __name__ == '__main__'`` condition.
+
+
+Configuration
+-------------
+
+The apidoc extension uses the following configuration values:
+
+.. confval:: apidoc_modules
+   :no-index:
+   :type: :code-py:`Sequence[dict[str, Any]]`
+   :default: :code-py:`()`
+
+   A list or sequence of dictionaries describing modules to document.
```

Comment:
On a side note, I could see the case for making config parameters applying to all modules. This would remove the need for repeating a lot of keys for multiple modules.

```
apidoc_module_config = {
        'followlinks': False,
        'separatemodules': False,
        'includeprivate': False,
        'noheadings': False,
}
apidoc_modules = {
        {'path': 'path/to/module'},
        {'path': 'path/to/other_module'},
        {'path': 'path/to/third_module', 'includeprivate': True}, # overriding global settings
}
```
Implementation-wise one would just merge the dicts `apidoc_module_config` and the individual config dicts.

But that could also be added later.

Reply:
Cc @chrisjsewell -- I just rebased the original PR so would prefer to have your input on larger design changes such as Tim suggests.

Reply:
Thanks @AA-Turner I'll have a look in the next few days